app/assets/javascripts/discourse/app/routes/review-index.js

```diff
@@ -82,10 +83,6 @@ export default DiscourseRoute.extend({
 
   deactivate() {
     this.messageBus.unsubscribe("/reviewable_claimed");
-    const channel = this.currentUser.enable_redesigned_user_menu
-      ? `/reviewable_counts/${this.currentUser.id}`
-      : "/reviewable_counts";
-    this.messageBus.unsubscribe(channel);
```

Comment:
This is a bug because we should not be unsubscribing here as this will unsubscribe the callback registered here

https://github.com/discourse/discourse/blob/ab58b0cffe5d8479037e9de346b3e323bf2e6b54/app/assets/javascripts/discourse/app/initializers/subscribe-user-notifications.js#L31-L38

The new test which I added will fail if we don't remove this line.